Issue-2238: Fixes the error messages on various forms.

sonvir249 opened the pull request:
## Describe your changes

Fixes the error message on signup and reset form.

## Write your issue number after "Fixes "

Fixes #2238

- [x] My PR is granular and targeted to one specific feature.
- [x] I took a screenshot or a video and attached to this PR if there is a UI change.

**Screenshots**
| Form Name | Current behavior  | Expected behavior |
| ------------- | ------------- | ------------- |
| Reset Password  |  ![Image](https://github.com/user-attachments/assets/7216cb0a-0b76-4918-a7d3-80843a5e5ee9)| ![Image](https://github.com/user-attachments/assets/bc1931f8-b587-48ec-8a57-25ac43243f55) |
| Signup form step 1  |![Image](https://github.com/user-attachments/assets/feaae740-d5be-4019-8dc0-141bac71b397)| ![Image](https://github.com/user-attachments/assets/d360bea0-e0cd-4566-aab9-e6cb16db231c)|
| Signup form step 2  |![Image](https://github.com/user-attachments/assets/940f3dc1-8337-433c-a50b-0cd26fd7e13d)| ![Image](https://github.com/user-attachments/assets/fbb8f9b0-20a0-40eb-959c-a2e6926e0f59)|




<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Error messages for email fields in Forgot Password and Register forms are now localized.
  - Surname validation now provides specific error messages distinct from first name validation.

- **Refactor**
  - Improved consistency in form validation schemas for better maintainability and readability.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Br0wnHammer commented on client/src/Pages/Auth/ForgotPassword.jsx:
```diff
@@ -185,7 +185,7 @@ const ForgotPassword = () => {
 							onChange={handleChange}
 							error={errors.email ? true : false}
-							helperText={errors.email}
+							helperText={t(errors.email)}
```
Any particular reason for this change? Also, can you share a UI SS.

sonvir249 replied:
@Br0wnHammer I added the detailed screenshots in the issue #2238. I've attached the SS in PR too.

Br0wnHammer commented on client/src/Pages/Auth/Register/StepTwo/index.jsx:
```diff
@@ -68,11 +68,14 @@ function StepTwo({ form, errors, onSubmit, onChange, onBack }) {
+								? t("authRegisterEmailRequired")
+								: errors.email.includes("valid email")
+									? t("authRegisterEmailInvalid")
+									: t(errors.email))
```
Same thing here.

sonvir249 replied:
@Br0wnHammer removed unnecessary changes.

Br0wnHammer commented on client/src/Validation/validation.js:
```diff
@@ -53,7 +66,7 @@ const passwordSchema = joi
 const credentials = joi.object({
 	firstName: nameSchema,
-	lastName: nameSchema,
+	lastName: surnameSchema,
```
A better name would be `lastnameSchema`.

sonvir249 replied:
@Br0wnHammer done.

Br0wnHammer commented on client/src/Validation/validation.js:
```diff
@@ -152,20 +165,21 @@ const monitorValidation = joi.object({
+	port: joi
+		.number()
+		.integer()
+		.min(1)
```
What is the change here?

sonvir249 replied:
@sonvir249 removed unnecessary change

Br0wnHammer commented on client/src/Validation/validation.js:
```diff
@@ -252,15 +266,16 @@ const statusPageValidation = joi.object({
+		}),
+		pagespeedApiKey: joi.string().allow("").optional(),
+	})
+	.unknown(true);
```
Again, please refrain from using formatting on the code that is not supported by us. Makes the diff larger than it is.

sonvir249 replied:
@Br0wnHammer removed unnecessary change

Br0wnHammer commented on client/src/Validation/validation.js:
```diff
@@ -377,5 +395,5 @@ export {
 	infrastructureMonitorValidation,
 	statusPageValidation,
-	logoImageValidation
+	logoImageValidation,
```
Same.

sonvir249 replied:
@Br0wnHammer removed unnecessary change

Br0wnHammer reviewed (changes requested):
Overall a UI SS is needed to review completely. Apart from that formatted code needs to omitted out of the PR. Larger diffs than required.

Owaiseimdad commented on client/src/Validation/validation.js:
```diff
@@ -11,7 +11,20 @@ const nameSchema = joi
+			"Name must contain only letters, spaces, apostrophes, or hyphens",
+	});
+
+const surnameSchema = joi
```
@sonvir249 , it looks like the same Joi schema is being repeated. Let's simplify this by renaming it to nameStringSchema. Also, instead of using name and surname, let's generalize it to something like characters or simply indicate that input is required. Please make sure to apply the DRY (Don't Repeat Yourself) principle wherever applicable.

sonvir249 replied:
@Br0wnHammer removed unnecessary change

Owaiseimdad commented on client/src/Pages/Auth/Register/StepTwo/index.jsx:
```diff
@@ -68,11 +68,14 @@ function StepTwo({ form, errors, onSubmit, onChange, onBack }) {
+								? t("authRegisterEmailRequired")
+								: errors.email.includes("valid email")
+									? t("authRegisterEmailInvalid")
+									: t(errors.email))
```
@sonvir249 does anything like t(errors.email) exists in local file and, is it even intented to be from translation or is it just errors.email?

sonvir249 replied:
@Owaiseimdad `errors.email` returns the `authRegisterEmailInvalid` and `authRegisterEmailRequired` and it should be passed in `t()` to print the translation string.

Owaiseimdad commented on client/src/Pages/Auth/ForgotPassword.jsx:
```diff
@@ -185,7 +185,7 @@ const ForgotPassword = () => {
 							onChange={handleChange}
 							error={errors.email ? true : false}
-							helperText={errors.email}
+							helperText={t(errors.email)}
```
same here, I don't see anything like errors.email in translation file, let me know if I am missing anything here.
@sonvir249

sonvir249 replied:
@Owaiseimdad `errors.email` returns the `authRegisterEmailInvalid` and `authRegisterEmailRequired` and it should be passed in `t()` to print the translation string.

Owaiseimdad reviewed (changes requested):
@sonvir249 please make your changes and test it locally and add screenshots for reference.

sonvir249 commented on client/src/Validation/validation.js:
```diff
@@ -252,15 +266,16 @@ const statusPageValidation = joi.object({
-	pagespeedApiKey: joi.string().allow("").optional(),
-})
-.unknown(true);
+const settingsValidation = joi
```
@Owaiseimdad Removed this formatting change.

sonvir249 commented:
> @sonvir249 please make your changes and test it locally and add screenshots for reference.

@Owaiseimdad @Owaiseimdad thanks for the review and detailed comments. I appreciate that.
I added all the details and screenshots in #2239 when I created the issue. As per you comment I've updated the PR description.

gorkem-bwl commented:
> > @sonvir249 please make your changes and test it locally and add screenshots for reference.
> 
> @Owaiseimdad @Owaiseimdad thanks for the review and detailed comments. I appreciate that. I added all the details and screenshots in #2239 when I created the issue. As per you comment I've updated the PR description.

Thanks @sonvir249. The screenshot is for the PR, not the issue, eg. reflects what you did rather than what the issue is. For this reason we are asking it to be added to the PR instead.

Br0wnHammer reviewed (approved):
LGTM! Thanks for the quick revisions and for explaining your approach to us. We appreciate your contributions.

Br0wnHammer commented:
@ajhollid this is fine for merging.

ajhollid commented:
Hey @sonvir249 , just a heads up we want to remove package-lock.json from the RP, not delete it from the project.

If you're not especially familiar with the process of ammending your git history it may be faster/easier to just open a new PR.  Thanks!

sonvir249 commented:
> Hey @sonvir249 , just a heads up we want to remove package-lock.json from the RP, not delete it from the project.
> 
> If you're not especially familiar with the process of ammending your git history it may be faster/easier to just open a new PR. Thanks!

@ajhollid I've updated current PR with develop branch. Please let me know if any other changes are needed. Thanks!
